include/wlr/types/wlr_pointer_gestures_v1.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_POINTER_GESTURES_V1_H
+#define WLR_TYPES_WLR_POINTER_GESTURES_V1_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_pointer.h>
+
+struct wlr_pointer_gestures_v1 {
+	struct wl_global *wl_global;
+	struct wl_list resources;
+	struct wl_list swipes; // wlr_pointer_gesture_swipe_v1::link
+	struct wl_list pinches; // wlr_pointer_gesture_pinch_v1::link
+
+	struct wl_listener display_destroy;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	void *data;
+};
+
+struct wlr_pointer_gesture_swipe_v1 {
+	struct wl_resource *resource;
+	struct wl_list link;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	void* data;
+};
+
+struct wlr_pointer_gesture_pinch_v1 {
+	struct wl_resource *resource;
+	struct wl_list link;
+
+	struct {
+		struct wl_signal destroy;
+	} events;
+
+	void* data;
+};
+
+void wlr_pointer_gesture_swipe_notify_begin(
+	struct wlr_pointer_gesture_swipe_v1 *swipe,
+	struct wl_resource *surface,
```

Comment:
We try not to use `wl_resource` in our APIs, because these are not type-checked to be `wl_surface` instances. Instead, the focused surface can be retrieved directly from the `wlr_seat`.